packages/components/src/select-control/index.js

```diff
@@ -92,21 +93,29 @@ export class SelectControl extends Component {
 	}
 
 	selectOption( option ) {
-		const { multiple, onChange, selected } = this.props;
-		const { query } = this.state;
+		const { multiple, selected } = this.props;
 		const newSelected = multiple ? [ ...selected, option ] : [ option ];
 
 		this.reset( newSelected );
 
+		const oldSelected = Array.isArray( selected )
+			? selected
+			: [ { key: selected } ];
+		const isSelected = findIndex( oldSelected, { key: option.key } );
+		if ( isSelected === -1 ) {
+			this.setNewValue( newSelected );
+		}
+	}
+
+	setNewValue( newValue ) {
+		const { onChange, selected, multiple } = this.props;
+		const { query } = this.state;
 		// Trigger a change if the selected value is different and pass back
 		// an array or string depending on the original value.
-		if ( Array.isArray( selected ) ) {
-			const isSelected = findIndex( selected, { key: option.key } );
-			if ( isSelected === -1 ) {
-				onChange( newSelected, query );
-			}
-		} else if ( selected !== option.key ) {
-			onChange( option.key, query );
+		if ( multiple || Array.isArray( selected ) ) {
+			onChange( newValue, query );
+		} else if ( newValue.length < 2 ) {
+			onChange( newValue.length > 0 ? newValue[ 0 ].key : '', query );
```

Comment:
I think this is all looking pretty good, but esp because this is a reused component could you cover some of these scenarios with some unit tests. I think that would really help reveal intent and also help us keep behaviour consistent when refactors in future come up.

Reply:
Yeah, I will re-visit test coverage, I started it Monday and then ran out of time, so scratched it. I will work on those.